tests/unit_tests/utils/test_general.py

```diff
@@ -206,3 +211,54 @@ def check_results(x_to_test, y_to_test, x_right, y_right, angle):
         gen.rotate(x_new_array.to(u.cm), y_new_array, angle_deg)
     with pytest.raises(u.core.UnitsError):
         gen.rotate(x_new_array, y_new_array, 30 * u.m)
+
+
+def test_convert_2D_to_radial_distr(caplog):
+
+    # Test normal functioning
+    max_dist = 100
+    num_bins = 100
+    step = max_dist / num_bins
+    xaxis = np.arange(-max_dist, max_dist, step)
+    yaxis = np.arange(-max_dist, max_dist, step)
+    x2d, y2d = np.meshgrid(xaxis, yaxis)
+    distance_to_center_2D = np.sqrt((x2d) ** 2 + (y2d) ** 2)
+
+    radial_edges, distance_to_center_1D = gen.convert_2D_to_radial_distr(
+        xaxis, yaxis, distance_to_center_2D, num_bins=num_bins, max_dist=max_dist
+    )
+    difference = radial_edges[:-1] - distance_to_center_1D
+
+    assert pytest.approx(difference[:-1], 1) == 0  # last value deviates
+
+    # Test warning in caplog
+    gen.convert_2D_to_radial_distr(
+        xaxis, yaxis, distance_to_center_2D, num_bins=4 * num_bins, max_dist=max_dist
+    )
+    msg = (
```

Comment:
It's really enough to check that just part of the message is printed (2-3 words) =)